/*
 * Copyright 2006 Sun Microsystems, Inc.  All rights reserved.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the
 * "Software"), to deal in the Software without restriction, including
 * without limitation the rights to use, copy, modify, merge, publish,
 * distribute, and/or sell copies of the Software, and to permit persons
 * to whom the Software is furnished to do so, provided that the above
 * copyright notice(s) and this permission notice appear in all copies of
 * the Software and that both the above copyright notice(s) and this
 * permission notice appear in supporting documentation.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
 * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
 * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
 * OF THIRD PARTY RIGHTS. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
 * HOLDERS INCLUDED IN THIS NOTICE BE LIABLE FOR ANY CLAIM, OR ANY SPECIAL
 * INDIRECT OR CONSEQUENTIAL DAMAGES, OR ANY DAMAGES WHATSOEVER RESULTING
 * FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
 * NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
 * WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 *
 * Except as contained in this notice, the name of a copyright holder
 * shall not be used in advertising or otherwise to promote the sale, use
 * or other dealings in this Software without prior written authorization
 * of the copyright holder.
 *
 */

diff -urp -x '*~' driver/lock.c driver~/lock.c
diff -urp -x '*~' driver/Makefile.in driver~/Makefile.in
--- driver/lock.c	2007-02-15 13:20:38.239047000 -0800
+++ driver~/lock.c	2007-02-23 15:56:09.614852000 -0800
@@ -378,6 +378,7 @@ fprintf(stderr,"Debug: ==> winatt_reset[
 /* cases for running GOK , or MAG or GOK+MAG */
 /* 6461887(P1) screen lock does not prevent access to other
 **  applications via 'alt-tab',
+** see 6519433 mem leaks in libgconf-2.so's gconf_client_get_default()
 */
 	client = gconf_client_get_default();
        
@@ -1530,11 +1531,14 @@ destroy_passwd_window (saver_info *si)
  }
 /*
 ** restore the prev one
+** 6520014(P3) mem. leak in lock.c
 */
 	if ( client && globalkey_text)
 	gconf_client_set_string ( client, \
         "/apps/metacity/global_keybindings/switch_windows", \
 	globalkey_text, NULL);
+	g_free(globalkey_text);
+	globalkey_text = NULL;
 }
